engine/engine.go

```diff
@@ -330,7 +334,8 @@ func (q *compatibilityQuery) Exec(ctx context.Context) (ret *promql.Result) {
 		return &promql.Result{Value: promql.String{V: e.Val, T: q.ts.UnixMilli()}}
 	}
 	ret = &promql.Result{
-		Value: promql.Vector{},
+		Value:    promql.Vector{},
+		Warnings: q.warns,
```

Comment:
I think we need to merge here with the warnings from the context maybe?

Reply:
Done